### PR TITLE
Add mqtt inbound endpoint ssl parameters

### PR DESCRIPTION
**Pull request by ThishaniLucas**
## Purpose
Adding mqtt inbound endpoint ssl parameters.
Fixed : wso2/product-ei#1516

### CODE REVIEW

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java [hunk 1]
```diff
@@ -2513,6 +2520,160 @@ protected void addStatisticsEnabledPropertyDescriptor(Object object) {
 				 null));
 	}
 
+	/**
+	 * This adds a property descriptor for the Transport MQTT Ssl Keystore Location feature.
+	 * <!-- begin-user-doc -->
+	 * <!-- end-user-doc -->
+	 * @generated NOT
+	 */
+	protected void addTransportMQTTSslKeystoreLocationPropertyDescriptor(Object object) {
```
Comment (sdkottegoda) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java]: There seems to be an issue in formatting.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java [hunk 2]
```diff
@@ -721,7 +729,28 @@ private void addParameterForInboundEndpoint(Map.Entry<String, String> paramEntry
                         } else {
                             executeSetValueCommand(INBOUND_ENDPOINT__CONTENT_TYPE, ContentType.JSON);
                         }
-                    }
+                    } else if (paramEntry.getKey().equals(InboundEndpointConstants.INBOUND_MQTT_SSL_KEYSTORE_LOCATION)) {
+	                 	executeSetValueCommand(INBOUND_ENDPOINT__TRANSPORT_MQTT_SSL_KEYSTORE_LOCATION, 
+	                 			paramEntry.getValue());                    	
+	                } else if (paramEntry.getKey().equals(InboundEndpointConstants.INBOUND_MQTT_SSL_KEYSTORE_TYPE)) {
```
Comment (sdkottegoda) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java]: Wouldn't it be clearer if we have a switch instead?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java (hunk 1 above)
Comment (ThishaniLucas) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java]: This code is auto-generated. Need to figure out some other way to fix the issue.

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java (hunk 2 above)
Comment (ThishaniLucas) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java]: Yes it definitely would. But the existing code base is having only if-else. Therefore, just followed the convention.

Comment (Drifftr) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java]: Shall we fix the formatting issues here?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/InboundEndpointTransformer.java [hunk 3]
```diff
@@ -964,6 +964,35 @@ private InboundEndpoint create(
                 addParameterForConfig(inboundEndpoint, InboundEndpointConstants.INBOUND_MQTT_RECONNECTION_INTERVAL,
                         visualInboundEndpoint.getInterval());
             }
+            if (StringUtils.isNotBlank(visualInboundEndpoint.getTransportMQTTSslKeystoreLocation())) {
```
Comment (Drifftr) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.persistence/src/org/wso2/developerstudio/eclipse/gmf/esb/internal/persistence/InboundEndpointTransformer.java]: Fix formatting issue here

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/model/esb.ecore [hunk 4]
```diff
@@ -1434,7 +1434,7 @@
     <eStructuralFeatures xsi:type="ecore:EAttribute" name="headersToExcludeInHash"
         eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EString"/>
     <eStructuralFeatures xsi:type="ecore:EAttribute" name="responseCodes" eType="ecore:EDataType http://www.eclipse.org/emf/2002/Ecore#//EString"
-        defaultValueLiteral=".*"/>
+        defaultValueLiteral="*"/>
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb/model/esb.ecore]: Default value should be corrected to ".*". Shall we add it back and generate again?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/EsbPackageImpl.java [hunk 5]
```diff
@@ -22099,7 +22169,7 @@ public void initializePackageContents() {
...
 		initEReference(getCacheMediator_MediatorFlow(), this.getMediatorFlow(), null, "mediatorFlow", null, 0, 1, CacheMediator.class, !IS_TRANSIENT, !IS_VOLATILE, IS_CHANGEABLE, IS_COMPOSITE, !IS_RESOLVE_PROXIES, !IS_UNSETTABLE, IS_UNIQUE, !IS_DERIVED, IS_ORDERED);
 		initEAttribute(getCacheMediator_HeadersToExcludeInHash(), ecorePackage.getEString(), "headersToExcludeInHash", null, 0, 1, CacheMediator.class, !IS_TRANSIENT, !IS_VOLATILE, IS_CHANGEABLE, !IS_UNSETTABLE, !IS_ID, IS_UNIQUE, !IS_DERIVED, IS_ORDERED);
-		initEAttribute(getCacheMediator_ResponseCodes(), ecorePackage.getEString(), "responseCodes", ".*", 0, 1, CacheMediator.class, !IS_TRANSIENT, !IS_VOLATILE, IS_CHANGEABLE, !IS_UNSETTABLE, !IS_ID, IS_UNIQUE, !IS_DERIVED, IS_ORDERED);
+		initEAttribute(getCacheMediator_ResponseCodes(), ecorePackage.getEString(), "responseCodes", "*", 0, 1, CacheMediator.class, !IS_TRANSIENT, !IS_VOLATILE, IS_CHANGEABLE, !IS_UNSETTABLE, !IS_ID, IS_UNIQUE, !IS_DERIVED, IS_ORDERED);
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/EsbPackageImpl.java]: Could you please revert ".*" change here?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java [hunk 6]
```diff
@@ -729,7 +737,28 @@ private void addParameterForInboundEndpoint(Map.Entry<String, String> paramEntry
...
```
Comment (prabushi) [inline plugins/org.wso2.developerstudio.eclipse.gmf.esb.diagram/src/org/wso2/developerstudio/eclipse/gmf/esb/diagram/custom/deserializer/InboundEndpointDeserializer.java]: Shall we fix the formatting issue before the "}"?

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/model/esb.ecore (hunk 4 above)
Comment (ThishaniLucas) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb/model/esb.ecore]: done

File: plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/EsbPackageImpl.java (hunk 5 above)
Comment (ThishaniLucas) [reply plugins/org.wso2.developerstudio.eclipse.gmf.esb/src/org/wso2/developerstudio/eclipse/gmf/esb/impl/EsbPackageImpl.java]: done

**Pull request merged**